docs/en/mavlink/index.md

```diff
@@ -0,0 +1,88 @@
+  These are supported by many flight stacks, ground stations, and MAVLink peripherals.
+  Flight stacks that use these definitions are more likely to interoperate.
+- [standard.xml](https://mavlink.io/en/messages/standard.html) — Definitions that are actually standard.
+  They are present on the vast majority of flight stacks and implemented in the same way.
+- [minimal.xml](https://mavlink.io/en/messages/minimal.html) — Definitions required by a minimal MAVLink implementation.
+
+The project also hosts [dialect XML definitions](https://mavlink.io/en/messages/#dialects), which contain MAVLink definitions that are specific to a flight stack or other stakeholder.
+
+The protocol relies on each end of the communication having a shared definition of what messages are being sent.
+What this means is that in order to communicate both ends of the communication must use libraries generated from the same XML definition.
+
+<!--
+The messages are sent over-the-wire in the "payload" of a [MAVLink packet](https://mavlink.io/en/guide/serialization.html#mavlink2_packet_format).
+In order to reduce the amount of information that must be sent, the packet does not include the message metadata, such as what fields are in the message and so on.
+Instead, the fields are serialized in a predefined order based on data size and XML definition order, and MAVLink relies on each end of the communication having a shared definition of what messages are being sent.
+The shared identity of the message is conveyed by the message id, along with a CRC ("`CRC_EXTRA`") that uniquely identifies the message based on its name and id, and the field names and types.
+The receiving end of the communication will discard any packet for which the message id and the `CRC_EXTRA` do not match.
+-->
+
+## PX4 and MAVLink
+
+PX4 releases build `common.xml` MAVLink definitions by default, for the greatest compatibility with MAVLink ground stations, libraries, and external components such as MAVLink cameras.
+In the `main` branch, these are included from `development.xml` on SITL, and `common.xml` for other boards.
+
+::: info
+To be part of a PX4 release, any MAVLink definitions that you use must be in `common.xml` (or included files such as `standard.xml` and `minimal.xml`).
+During development you can use definitions in `development.xml`.
+You will need to work with the [MAVLink team](https://mavlink.io/en/contributing/contributing.html) to define and contribute these definitions.
+:::
+
+PX4 includes the [mavlink/mavlink](https://github.com/mavlink/mavlink) repo as a submodule under [/src/modules/mavlink](https://github.com/PX4/PX4-Autopilot/tree/main/src/modules/mavlink).
+This contains XML definition files in [/mavlink/messages/1.0/](https://github.com/mavlink/mavlink/blob/master/message_definitions/v1.0/).
+
+The build toolchain generates the MAVLink 2 C header files at build time.
```

Comment:
```suggestion
The build toolchain generates the MAVLink 1 and 2 C header files at build time.
```

Actually, I only see the generation for MAVLink 2, as you say. However, PX4 does also talk v1, so I'm not sure how that works.

https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/mavlink/CMakeLists.txt#L64

Reply:
UPshot, this bit of the text is correct AFAIK.